libbeat/outputs/elasticsearch/url.go

```diff
@@ -114,3 +114,18 @@ func makePath(index string, docType string, id string) (string, error) {
 	}
 	return path, nil
 }
+
+func parseProxyURL(raw string) (*url.URL, error) {
```

Comment:
added todo item. The HTTP connection/configurations should be split and generalized for all beats. e.g. heartbeat and metricbeat also have http clients with similar settings.